base/set.jl

```diff
@@ -383,14 +383,30 @@ false
 """
 function allunique(C)
     seen = Set{eltype(C)}()
-    for x in C
-        if in(x, seen)
-            return false
+
+    x = iterate(C)
+    for i in OneTo(1000)
+        if x === nothing
```

Comment:
Seems like this `if-else` block can be removed by just adding `x === nothing && return true` above the `for` loop?

Reply:
I think we need this check inside the loop for cases when the iterator has less than 1000 elements.

Reply:
Ah, obviously.